symengine/polynomial.cpp

```diff
@@ -20,11 +20,15 @@ UnivariateIntPolynomial::UnivariateIntPolynomial(
     const RCP<const Symbol> &var, const std::vector<integer_class> &v)
     : var_{var}
 {
-    for (unsigned int i = 0; i < v.size(); i++)
-        if (v[i] != 0)
-            dict_add_term(dict_, v[i], i);
-    degree_ = v.size() - 1;
-    SYMENGINE_ASSERT(is_canonical(degree_, dict_))
```

Comment:
It's not necessary, but having a function just for adding terms to the dictionary did not make sense. Also, checking if it was canonical was not needed again, the function ensures that the constructed polynomial is canonical.